### PR TITLE
feat(scheduler-alpha): target properties override

### LINKED ISSUE #27545: (aws-scheduler-alpha): Override target properties in schedule
### Describe the feature

If customers wish to reuse the same target in multiple schedules, they should be able to override target properties like input, maximumRetryAttempts and maximumEventAge when creating a Schedule:

```ts

const target = new targets.LambdaInvoke(props.func, {
    input: ScheduleTargetInput.fromObject({
        "payload": "useful"
    }),
});

const schedule1 = new Schedule(this, 'Schedule', {
    scheduleExpression: ScheduleExpression.cron({ day: '20' }),
    target,
});

const schedule2 = new Schedule(this, 'Schedule2', {
    scheduleExpression: ScheduleExpression.cron({ day: '5' })
    target,
    targetOverrides: {
        input: ScheduleTargetInput.fromText("Overriding Target Input")
    }
});
```

See https://github.com/aws/aws-cdk-rfcs/blob/main/text/0474-event-bridge-scheduler-l2.md#overriding-target-properties

### Use Case

It will be useful for customers who wish to reuse the same target in multiple schedules. 

### Proposed Solution

Add property `targetOverrides` to `ScheduleProps` that is being used during creation of `CfnSchedule`. 

```ts
export interface TargetOverrideProps {
  readonly input?: ScheduleTargetInput;
  readonly maximumEventAge?: Duration;
  readonly maximumRetryAttempts?: number;
}
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.99.1

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by lpizzinidev**
Allows to override a `Schedule` target's properties.
Supported properties: `input`, `maxEventAge`, and `retryAttempts`.

Example:
```ts
declare const target: targets.LambdaInvoke;

const oneTimeSchedule = new Schedule(this, 'Schedule', {
    schedule: ScheduleExpression.rate(cdk.Duration.hours(12)),
    target,
    targetOverrides: {
        input: ScheduleTargetInput.fromText("Overriding Target Input"),
        maxEventAge: Duration.seconds(180),
        retryAttempts: 5,
    },
});
```

Closes #27545.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by lpizzinidev** [comment]
Exemption Request.
Not sure why this got tagged `pr/needs-cli-test-run`.

File: packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts [hunk 1]
```diff
@@ -30,6 +33,54 @@ export interface ISchedule extends IResource {
...
+   *
+   * @default - The target's input is used.
+   */
+  readonly input?: ScheduleTargetInput;
+  /**
+   * The maximum amount of time, in seconds, to continue to make retry attempts.
+   *
+   * @default - The target's maximumEventAgeInSeconds is used.
+   */
+  readonly maximumEventAge?: Duration;
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts:51]: it's called `maxEventAge` in `schedule-targets`, lets call it that here too

File: packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts [hunk 2]
```diff
@@ -30,6 +33,54 @@ export interface ISchedule extends IResource {
...
+export interface ScheduleTargetProps {
+  /**
+   * The text, or well-formed JSON, passed to the target.
+   *
+   * If you are configuring a templated Lambda, AWS Step Functions, or Amazon EventBridge target,
+   * the input must be a well-formed JSON. For all other target types, a JSON is not required.
+   *
+   * @default - The target's input is used.
+   */
+  readonly input?: ScheduleTargetInput;
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts:69]:
```suggestion
  readonly input?: ScheduleTargetInput;

```

File: packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts [hunk 3]
```diff
@@ -30,6 +33,54 @@ export interface ISchedule extends IResource {
...
+  /**
+   * The maximum amount of time, in seconds, to continue to make retry attempts.
+   *
+   * @default - The target's maximumEventAgeInSeconds is used.
+   */
+  readonly maximumEventAge?: Duration;
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts:75]:
```suggestion
  readonly maximumEventAge?: Duration;

```

File: packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts [hunk 4]
```diff
@@ -30,6 +33,54 @@ export interface ISchedule extends IResource {
...
+  /**
+   * The maximum number of retry attempts to make before the request fails.
+   *
+   * @default - The target's maximumRetryAttempts is used.
+   */
+  readonly maximumRetryAttempts?: number;
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts:81]:
```suggestion
  readonly maxRetryAttempts?: number;
```

**Review by kaizencc — CHANGES_REQUESTED** [review]
Thanks @lpizzinidev! Super minor comments, otherwise i think we are gtg

File: packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts (hunk 4 above)
Comment (lpizzinidev) [reply packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts]: Renamed to `retryAttempts` as in [target](https://github.com/aws/aws-cdk/blob/fc3be313daf40bbf12b55f0b3ed3027505ca3a87/packages/%40aws-cdk/aws-scheduler-targets-alpha/lib/target.ts#L62) class.

File: packages/@aws-cdk/aws-scheduler-alpha/README.md [hunk 5]
```diff
@@ -256,8 +256,22 @@ const target = new targets.LambdaInvoke(fn, {
 
 ## Overriding Target Properties 
 
-TODO: Not yet implemented. See section in [L2 Event Bridge Scheduler RFC](https://github.com/aws/aws-cdk-rfcs/blob/master/text/0474-event-bridge-scheduler-l2.md)
+If you wish to reuse the same target in multiple schedules, you can override target properties like `input`, 
+`maximumRetryAttempts` and `maximumEventAgeInSeconds` when creating a Schedule using the `targetOverrides` parameter:
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-scheduler-alpha/README.md:260]:
```suggestion
`retryAttempts` and `maxEventAge` when creating a Schedule using the `targetOverrides` parameter:
```

File: packages/@aws-cdk/aws-scheduler-alpha/README.md [hunk 6]
```diff
@@ -256,8 +256,22 @@ const target = new targets.LambdaInvoke(fn, {
...
 
+```ts
+declare const target: targets.LambdaInvoke;
+
+const oneTimeSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(cdk.Duration.hours(12)),
+    target,
+    targetOverrides: {
+        input: ScheduleTargetInput.fromText("Overriding Target Input"),
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-scheduler-alpha/README.md:269]:
```suggestion
        input: ScheduleTargetInput.fromText('Overriding Target Input'),
```

File: packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts [hunk 7]
```diff
@@ -222,9 +263,11 @@ export class Schedule extends Resource implements ISchedule {
       target: {
         arn: targetConfig.arn,
         roleArn: targetConfig.role.roleArn,
-        input: targetConfig.input?.bind(this),
+        input: props.targetOverrides?.input ?
+          props.targetOverrides?.input?.bind(this) :
+          targetConfig.input?.bind(this),
         deadLetterConfig: targetConfig.deadLetterConfig,
-        retryPolicy: targetConfig.retryPolicy,
+        retryPolicy: retryPolicy.maximumEventAgeInSeconds || retryPolicy.maximumRetryAttempts ? retryPolicy : undefined,
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts:270]: I don't like this, it's unnecessarily complicated. And what if we add a prop to `retryPolicy` in the future? Lets add a function that creates a retryPolicy but returns undefined if no properties inside the object are defined

**Review by kaizencc — CHANGES_REQUESTED** [review]
One last nit @lpizzinidev

**Review by kaizencc — DISMISSED** [review]
Thanks!

**Review by kaizencc — APPROVED** [review]

**Pull request merged**